enterprise/server/cmd/ci_runner/main.go

```diff
@@ -936,19 +934,29 @@ func (r *buildEventReporter) Printf(format string, vals ...interface{}) {
 type invocationLog struct {
 	lockingbuffer.LockingBuffer
 	writer        io.Writer
-	writeListener func(b []byte)
+	writeListener func(s string)
 }
 
 func newInvocationLog() *invocationLog {
-	invLog := &invocationLog{writeListener: func(b []byte) {}}
+	invLog := &invocationLog{writeListener: func(s string) {}}
 	invLog.writer = io.MultiWriter(&invLog.LockingBuffer, os.Stderr)
 	return invLog
 }
 
 func (invLog *invocationLog) Write(b []byte) (int, error) {
-	invLog.writeListener(b)
-	n, err := invLog.writer.Write(b)
-	return n, err
+	output := string(b)
+
+	redacted, err := redact.RedactCommand(output)
```

Comment:
`RedactCommand` parses all input using `shlex`, but invocations contain a lot more output than just commands. Are there specific regex-replace rules we could apply here instead?

Reply:
I considered that too, but thought I'd just stick with the existing logic for simplicity. Are you concerned about the performance implications?

Reply:
`shlex.Split()` doesn't handle arbitrary text - it's a shell tokenizer. For example, `shlex.Split("Hello 'w o r l d'")` returns `[]string{"Hello", "w o r l d"}`. `shlex.Split(" ' ")` fails because it's not valid shell code (there's an unterminated closing quote).

Reply:
Refactored